subsys/bluetooth/controller/ll_sw/ull_adv_aux.c

```diff
@@ -775,42 +784,29 @@ uint8_t ull_adv_aux_hdr_set_clear(struct ll_adv_set *adv,
 	}
 
 	/* ADI */
-	{
-		struct pdu_adv_adi *pri_adi, *sec_adi;
-		uint16_t did = UINT16_MAX;
-
-		pri_dptr -= sizeof(struct pdu_adv_adi);
-		sec_dptr -= sizeof(struct pdu_adv_adi);
-
-		pri_adi = (void *)pri_dptr;
-		sec_adi = (void *)sec_dptr;
-
-		if (pri_hdr_prev.adi) {
-			struct pdu_adv_adi *pri_adi_prev;
+	if (pri_hdr_prev.adi) {
+		pri_dptr_prev -= sizeof(struct pdu_adv_adi);
+	}
+	if (sec_hdr_prev.adi) {
+		sec_dptr_prev -= sizeof(struct pdu_adv_adi);
+	}
 
-			pri_dptr_prev -= sizeof(struct pdu_adv_adi);
-			sec_dptr_prev -= sizeof(struct pdu_adv_adi);
+	pri_dptr -= sizeof(struct pdu_adv_adi);
+	sec_dptr -= sizeof(struct pdu_adv_adi);
 
-			memcpy(pri_dptr, pri_dptr_prev,
-			       sizeof(struct pdu_adv_adi));
-			memcpy(sec_dptr, sec_dptr_prev,
-			       sizeof(struct pdu_adv_adi));
+	pri_adi = (void *)pri_dptr;
+	sec_adi = (void *)sec_dptr;
 
-			pri_adi_prev = (void *)pri_dptr_prev;
-			did = sys_le16_to_cpu(pri_adi_prev->did);
-		} else {
-			pri_adi->sid = adv->sid;
-			sec_adi->sid = adv->sid;
-		}
+	pri_adi->sid = adv->sid;
+	sec_adi->sid = adv->sid;
 
-		did++;
+	did = did_unique[adv->sid]++;
```

Comment:
So it is the DID of a SID that needs to be unique? I.e. a set with SID = 0 and a set with SID = 1 can both have DID = 2 and be compliant? 

I suggest we add a description here stating that, as well as stating that this value is in fact 12 bits and may overflow without any negative consequences, e.g. something like


```suggestion
	/* The DID for a specific SID shall be unique. 
	 * The DID is 12 bits and may overflow without any negative consequences 
	 */
	did = did_unique[adv->sid]++;
```